File: src/main/java/com/kishlaly/ta/analyze/testing/sl/StopLossVolatileKeltnerMiddle.java
package com.kishlaly.ta.analyze.testing.sl;

import com.kishlaly.ta.model.SymbolData;
import com.kishlaly.ta.model.indicators.KeltnerJava;
import com.kishlaly.ta.utils.IndicatorUtils;

public class StopLossVolatileKeltnerMiddle extends StopLossStrategy {

    public StopLossVolatileKeltnerMiddle() {
        super(null, true);
    }

    @Override
    public double calculate(SymbolData data, int signalIndex) {
        KeltnerJava keltner = IndicatorUtils.buildKeltnerChannels(data.symbol, data.quotes).get(signalIndex);
        return keltner.getMiddle();
    }

    @Override
    public String toString() {
        return "SL volatile Keltner middle";
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/testing/sl/StopLossVolatileKeltnerMiddle.kt
package com.kishlaly.ta.analyze.testing.sl

import com.kishlaly.ta.model.SymbolData
import com.kishlaly.ta.utils.IndicatorUtils

class StopLossVolatileKeltnerMiddle : StopLossStrategy(isVolatile = true) {
    override fun calculate(data: SymbolData, signalIndex: Int): Double {
        return IndicatorUtils.buildKeltnerChannels(data.symbol, data.quotes)[signalIndex].middle
    }

    override fun toString() = "SL volatile Keltner middle"

}